### ISSUE TITLE
storage management?

### ISSUE THREAD

**Issue by KlavsKlavsen** [pravega/zookeeper-operator#136]
A ZookerCluster takes up 20GB og space per instance - ie. 60GB in total.. I can't find anywhere that documents how to specify a different disk size? 60GB is a rather large size..
and how can this be resized (up/down) ?

**Comment by KlavsKlavsen** [comment]
The helm chart for ZookeeperClsuter - also does NOT allow for specifying amount of disk space allocated :(

**Comment by spiegela** [comment]
@KlavsKlavsen You're correct that the Helm chart in this project is too simple.  We really only use it as an example chart implementation as our other projects are creating their own charts, but I think that it should be expanded to include more of the available options.

The ZK Operator does support customizing a number of storage options for the ZookeeperCluster instances. The [persistence spec API](https://github.com/pravega/zookeeper-operator/blob/master/pkg/apis/zookeeper/v1beta1/zookeepercluster_types.go#L358) represents the settable options for the volume claim template used.

**Comment by KlavsKlavsen** [comment]
So I can set:
```
PersistentVolumeClaimSpec.Resources.Requests.ResourceStorage: '8Gi'
``` 
? a yaml example would be nice :)

**Comment by spiegela** [comment]
Yes, that's right.  Here's a full YAML example:

```
apiVersion: "zookeeper.pravega.io/v1beta1"
kind: "ZookeeperCluster"
metadata:
  name: example
spec:
  replicas: 3
  image:
    repository: pravega/zookeeper
    tag: 0.2.5
    pullPolicy: {{ .Values.pullPolicy }}
  labels:
    app: zookeeper
    component: zk
  persistence:
    volumeReclaimPolicy: Retain
    spec:
      storageClassName: my-storageclass
      resources:
        requests:
          storage: 8Gi
```

The `persistence.spec` section uses the `PersistentVolumeClaimSpec` API, so any fields in that should be passed to the volumes:  `AccessModes`, `Selector`, etc. However, they must be compatible with a `StatefulSet` type, which the ZK Operator uses as a secondary resource.

**Comment by KlavsKlavsen** [comment]
Thank you.. Would you appreciate a PR for extending the helm chart in this repo - so it allows setting storageclass, storage-size and image details ? and perhaps allow for a hash of labels - all of them with no defaults set (so it uses what the operator sets when none is given)

**Comment by anishakj** [comment]
Fixed by PR #173
